mods/simple_service.js

```diff
@@ -0,0 +1,126 @@
+    return title.replace(/ /g, '_');
+}
+
+function SimpleService(options) {
+    options = options || {};
+    this.spec = {
+        paths: options.paths
+    };
+
+    this.exports = this.processSpec(this.spec);
+}
+
+var headerWhitelist = ['accept', 'accept-language', 'accept-language'];
+function filterHeaders(headers) {
+    var res = {};
+    headerWhitelist.forEach(function(name) {
+        if (headers[name]) {
+            res[name] = headers[name];
+        }
+    });
+    return res;
+}
+
+SimpleService.prototype.processSpec = function(spec) {
+    var self = this;
+    var operations = {};
+    var resources = [];
+
+    Object.keys(spec.paths).forEach(function(path) {
+        var pathObj = spec.paths[path];
+        Object.keys(pathObj).forEach(function(method) {
+            var conf = pathObj[method];
+            conf.operationId = method + '_' + path;
+            var storageUriTemplate;
+            if (conf.storage) {
+                if (!conf.storage.bucket_request.uri) {
+                    throw new Error('Broken config: expected storage.bucket_request.uri for ' + path);
+                }
+                storageUriTemplate = new URI(conf.storage.item_request.uri, {}, true);
+                resources.push(conf.storage.bucket_request);
+            }
+            var backendUriTemplate = new URI(conf.backend_request.uri, {}, true);
+            operations[conf.operationId] = function(restbase, req) {
+                var rp = req.params;
+                if (rp.key) {
+                    rp.key = normalizeTitle(rp.key);
+                }
+
+                function backendRequest() {
+                    var beReq = {
+                        uri: backendUriTemplate.toString({
+                            params: req.params
+                        }),
+                        // TODO: be more selective / only configure a whitelist of
+                        // headers
+                        headers: filterHeaders(req.headers),
+                        method: method,
+                        body: req.body,
+                    };
+                    return restbase.request(beReq);
+                }
+
+                function regenerateAndSave() {
+                    // Fall back to the backend service
+                    return backendRequest()
+                    .then(function(res) {
+                        // store the result
+                        return restbase.put({
+                            uri: storageUriTemplate.expand(req.params),
+                            headers: res.headers,
+                            body: res.body,
+                        })
+                        .then(function(storeRes) {
+                            res.headers.etag = storeRes.headers.etag;
+                            return res;
+                        });
+                    });
+                }
+
+                if (conf.storage) {
+                    if (conf.storage['no-cache_refresh']
```

Comment:
The `no-cache` part reflects the header. I agree that it doesn't look pretty, but at least it's consistent with our policy of using _ as a delimiter between components.